src/Codec/Beam.hs

```diff
@@ -266,6 +285,39 @@ encodeExports builder =
       pack32 (atomTable builder ! name) <> pack32 arity <> pack32 labelId
 
 
+encodeLiterals :: Builder -> BS.ByteString
+encodeLiterals builder =
+  uncompressedSize <> compressed
+
+    where
+      compressed =
+        Zlib.compress (count <> encodedLiterals)
+
+      count =
+        pack32 $ length (litTable builder)
+
+      encodedLiterals =
+        BS.concat $ map encodeLiteral (litTable builder)
+
+      uncompressedSize =
+        pack32 $ BS.length (count <> encodedLiterals)
```

Comment:
It seems like `count <> encodedLiterals` should be extracted since it's used twice. Also, I'm not a fan of having all of these names: `encodedLiterals`, `encodeLiterals`, and `encodeLiteral` - they're too similar to keep track of. I don't have any suggestions off the top of my head, but Slack me if you wanna figure something out together.